db/fixtures/electoral_commission_parties.rb

```diff
@@ -0,0 +1,136 @@
+require "open-uri"
+require "csv"
+
+# Example of use in terminal
+# $ bundle exec rails c
+# > require "./db/fixtures/electoral_commission_parties"
+# > ElectoralCommissionParties.download
+#
+# > ecdata =  ElectoralCommissionParties.new
+# > ecdata.find_by_name("Liberal Democrat")
+# => {"ECRef"=>"PP90", "RegulatedEntityName"=>"Liberal Democrats", ...
+#
+# > uni = ecdata.unique_entities
+# > uni["Labour Party"]
+# => {:regulated_entity_name=>"Labour Party", :joint_description=>"Labour and Co-operative Party" ...
+# > uni["Co-operative Party"]
+# => {:regulated_entity_name=>"Co-operative Party", :joint_description=>"Labour and Co-operative Party" ...
+# > uni["Conservative and Unionist Party"]
+# => {: regulated_entity_name=>"Conservative and Unionist Party", :descriptions => ["Conservatives", "NI Conservatives"
+
+module Db
+  module Fixtures
+    class ElectoralCommissionParties
+      include Enumerable
+
+      # wikipedia page at https://en.wikipedia.org/wiki/List_of_political_parties_in_the_United_Kingdom#cite_note-1
+      # starts with http://search.electoralcommission.org.uk/Search/Registrations?currentPage=1&rows=10&sort=RegulatedEntityName&order=asc&et=pp&et=ppm&register=gb&register=ni&register=none&regStatus=registered
+      # we have tacked on getDescriptions=true and adapted the link to point to the CSV download,
+      # and not the html search page
+
+      URL = "http://search.electoralcommission.org.uk/api/csv/Registrations?sort=RegulatedEntityName&order=asc&et=pp&et=ppm&register=gb&register=ni&register=none&regStatus=registered&getDescriptions=true"
+      FILE = File.expand_path("./electoral_commission_parties.csv", __dir__)
+
+      class << self
+        def download
+          # ec_parties_data_as_csv = URI.open(URL).read ruby 3.0 syntax
+          ec_parties_data_as_csv = open(URL).read
+          File.write(FILE, ec_parties_data_as_csv)
+        end
+      end
+
+      # The EC has a concept of unique entities, so one parliamentary entity has a unique name,
+      # even though it may have more than one registration (GB and NI) for that single name.
+      # This returns a hash of those entities, with all their registered descriptions, and all registrations.
+      # Also, joint descriptions are collected, so that parties in permanent alliance can be tracked.
+      # Labour and Co-op for instance are returned as distinct
+      # entities, which can be found to work together by matching their joint description.
+      def unique_entities
+        return @unique_entities if defined?(@unique_entities)
+        @unique_entities = each_with_object({}) do  |p, result|
```

Comment:
I'd suggest using a more descriptive variable name than `p`, especially given that `p` is [already defined by `Kernel`](https://rubydoc.info/stdlib/core/Kernel#p-instance_method).